### PR TITLE
Publish Preview Package

### PR DESCRIPTION
**Pull request by wbaldoumas**
## Description

- Include `LICENSE` in packaged library
- Publish a preview version of the package for a commit when a preview tag is pushed for a commit

This change is the first step for addressing issue https://github.com/wayfair-incubator/wayfair-text-json/issues/11.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/wayfair-text-json/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: .github/workflows/publish-preview.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Restore Dependencies
+      run: cd src && dotnet restore
+    - name: Build
+      run: cd src && dotnet build --configuration Release --no-restore
+    - name: Set VERSION variable from tag
+      run: echo "VERSION=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
+    - name: Pack NuGet
+      run: cd src && dotnet pack --no-build --configuration Release /p:Version=${VERSION} --output .
+    - name: Publish NuGet
+      run: cd src && dotnet nuget push *.nupkg --source https://api.nuget.org/v3/index.json --api-key ${NUGET_KEY} --skip-duplicate
```
Comment (jdhmtl) [inline .github/workflows/publish-preview.yml:32]: Would it make more sense to provide a path to the solution file rather than using `cd`?

Comment (wbaldoumas) [reply .github/workflows/publish-preview.yml:32]: Let me give that a try. I think another idea would be to set the working directory to `src` once, if we can do that in GitHub Actions.

**Comment by draco2003** [comment]
Did we investigate existing Github Actions for Nuget publishing? 

Example (not an endorsement):
https://github.com/marketplace/actions/publish-nuget

**Comment by wbaldoumas** [comment]
> Did we investigate existing Github Actions for Nuget publishing?
> 
> Example (not an endorsement): https://github.com/marketplace/actions/publish-nuget

We will be using that for the full release. Unfortunately that action does not provide an easy to use way of publishing preview versions based on when I worked with it in the past.

**Review by jdhmtl — APPROVED** [review]

**Comment by draco2003** [comment]
> > Did we investigate existing Github Actions for Nuget publishing?
> > Example (not an endorsement): https://github.com/marketplace/actions/publish-nuget
> 
> We will be using that for the full release workflow. Unfortunately that action does not provide an easy to use way of publishing preview versions based on when I worked with it in the past.

Sounds like a good upstream contribution. Thanks for the context.

**Pull request merged**